File: GameEngine_IOS/GameEngine/Models/ComplexEntities/Entity.swift
import Foundation

/**
 * Represents one entity in the 3D world
 */
public class Entity  {
    
    /**
     * Keys: Have the name of the group The name of the material group for
     * instance harm
     */
    var genericEntity : GenericEntity;
    
    /*Position where the entity is*/
    var  position : Vector3f
    
    /*Rotation (X-axle) of the 3D model */
    var  rotX : Float;
    
    /*Rotation (Y-axle) of the 3D model */
    var  rotY : Float
    /*Rotation (Z-axle) of the 3D model */
    var rotZ : Float
    
    /*Scale of the model*/
    var scale : Float
    
    
    /**
     * Initiator of the entity to be render in the 3D world
     *
     * @param genericEntity  Reference to a generic entity with the description of the model
     * @param position       Position where the model should be render
     * @param rotX           Rotation of the model in the X axle
     * @param rotY           Rotation of the model in the Y axle
     * @param rotZ           Rotation of the model in the Z axle
     * @param scale          Scale of the model
     */
    public init(genericEntity : GenericEntity, position : Vector3f, rotX : Float, rotY : Float, rotZ : Float, scale : Float) {
        self.genericEntity = genericEntity;
        self.position = position;
        self.rotX = rotX;
        self.rotY = rotY;
        self.rotZ = rotZ;
        self.scale = scale;
    }
    
    
    /**
     * Increases the position of the model using for that the specified components
     *
     * @param dx	X component to be increase
     * @param dy	Y component to be increase
     * @param dz	Z component to be increase
     */
    public func increasePosition(dx : Float, dy : Float, dz : Float) {
        self.position.x += dx;
        self.position.y += dy;
        self.position.z += dz;
    }
    
    /**
     * Increases the rotation of the model using for that the specified components
     * @param dx	X component to be increase
     * @param dy	Y component to be increase
     * @param dz	Z component to be increase
     */
    public func increaseRotation(dx : Float, dy : Float, dz : Float) {
        self.rotX += dx;
        self.rotY += dy;
        self.rotZ += dz;
    }
}

File: GameEngine_PC/GameEngineOpenGL3_2/src/main/java/com/dferreira/gameEngine/models/complexEntities/Entity.java
package com.dferreira.gameEngine.models.complexEntities;

import com.dferreira.commons.Vector3f;

/**
 * Represents one entity in the 3D world
 */
public class Entity {

	/**
	 * Keys: Have the name of the group The name of the material group for
	 * instance harm
	 */
	private final GenericEntity genericEntity;

	/* Position where the entity is */
	private Vector3f position;

	/* Rotation of the 3D model */
	private float rotX, rotY, rotZ;

	/* Scale of the model */
	private float scale;

	/**
	 * 
	 * Constructor of the entity to be render in the 3D world
	 * 
	 * @param genericEntity
	 *            reference to a generic entity with the description of the
	 *            model
	 * @param position
	 *            position where the model should be render
	 * @param rotX
	 *            Rotation of the model in the X axle
	 * @param rotY
	 *            Rotation of the model in the Y axle
	 * @param rotZ
	 *            Rotation of the model in the Z axle
	 * @param scale
	 *            Scale of the model
	 * 
	 */
	public Entity(GenericEntity genericEntity, Vector3f position, float rotX, float rotY, float rotZ, float scale) {
		super();
		this.genericEntity = genericEntity;
		this.position = position;
		this.rotX = rotX;
		this.rotY = rotY;
		this.rotZ = rotZ;
		this.scale = scale;
	}

	/**
	 * Increases the position of the model using for that the specified
	 * components
	 * 
	 * @param dx
	 *            X component to be increase
	 * @param dy
	 *            Y component to be increase
	 * @param dz
	 *            Z component to be increase
	 */
	public void increasePosition(float dx, float dy, float dz) {
		this.position.x += dx;
		this.position.y += dy;
		this.position.z += dz;
	}

	/**
	 * Increases the rotation of the model using for that the specified
	 * components
	 * 
	 * @param dx
	 *            X component to be increase
	 * @param dy
	 *            Y component to be increase
	 * @param dz
	 *            Z component to be increase
	 */
	public void increaseRotation(float dx, float dy, float dz) {
		this.rotX += dx;
		this.rotY += dy;
		this.rotZ += dz;
	}

	/**
	 * @return the position of the model
	 */
	public Vector3f getPosition() {
		return position;
	}

	/**
	 * 
	 * @param position
	 *            The position to be set
	 */
	public void setPosition(Vector3f position) {
		this.position = position;
	}

	/**
	 * @return the rotation of the entity in the X axle
	 */
	public float getRotX() {
		return rotX;
	}

	/**
	 * @param rotX
	 *            the rotation of the entity in the X axle
	 */
	public void setRotX(float rotX) {
		this.rotX = rotX;
	}

	/**
	 * @return the rotation of the entity in the Y axle
	 */
	public float getRotY() {
		return rotY;
	}

	/**
	 * @param rotY
	 *            the rotation of the entity in the Y axle
	 */
	public void setRotY(float rotY) {
		this.rotY = rotY;
	}

	/**
	 * @return the rotation of the entity in the Z axle
	 */
	public float getRotZ() {
		return rotZ;
	}

	/**
	 * @param rotZ
	 *            the rotation of the entity in the Z axle
	 */
	public void setRotZ(float rotZ) {
		this.rotZ = rotZ;
	}

	/**
	 * @return the scale the scale of the entity
	 */
	public float getScale() {
		return scale;
	}

	/**
	 * @param scale
	 *            the scale to set
	 */
	public void setScale(float scale) {
		this.scale = scale;
	}

	/**
	 * @return the genericEntity
	 */
	public GenericEntity getGenericEntity() {
		return genericEntity;
	}
}
